app/models/host/base.rb

```diff
@@ -246,7 +246,11 @@ def overwrite=(value)
     end
 
     def primary_interface
-      get_interface_by_flag(:primary)
+      # Returns a dummy object if host is unmanaged and without interfaces.
+      # Unmanaged hosts still have the interface requirement due to delegations and certain validations.
+      # A refactoring that allows unmanaged hosts to exist without interfaces would be welcome.
+      return get_interface_by_flag(:primary) if managed || interfaces.detect(&:primary)
+      self.interfaces.build(:primary => true, :provision => true, :managed => false, :type => 'Nic::Managed')
```

Comment:
Ah, I see your previous comment now. What was the reason for not using `build_required_interfaces` here too?